src/Windows/Avalonia.Win32/WindowImpl.AppWndProc.cs

```diff
@@ -467,6 +468,41 @@ protected virtual unsafe IntPtr AppWndProc(IntPtr hWnd, uint msg, IntPtr wParam,
                 case WindowsMessage.WM_KILLFOCUS:
                     LostFocus?.Invoke();
                     break;
+
+                case WindowsMessage.WM_INPUTLANGCHANGE:
+                    {
+                        UpdateInputMethod(lParam);
+                        // call DefWindowProc to pass to all children
+                        break;
+                    }
+                case WindowsMessage.WM_IME_SETCONTEXT:
+                    {
+                        // TODO if we implement preedit, disable the composition window:
```

Comment:
Technically preedit and other advanced features are already in the API: https://github.com/AvaloniaUI/Avalonia/blob/d2347bf0f75434ae9dfc9feb1045d790dd8ed0bd/src/Avalonia.Input/TextInput/ITextInputMethodClient.cs#L27-L51

but we have this chicken and egg problem: platform code doesn't handle preedit and textbox doesn't support it as well.

Reply:
I guess we need some control in the control catalog or sandbox to test the platform support.